src/huggingface_hub/keras_mixin.py

```diff
@@ -16,6 +20,129 @@
 
 logger = logging.getLogger(__name__)
 
+if is_tf_available():
+    import tensorflow as tf
+
+
+def _extract_hyperparameters_from_keras(model):
+    if model.optimizer is not None:
+        hyperparameters = dict()
+        hyperparameters["optimizer"] = model.optimizer.get_config()
+        hyperparameters[
+            "training_precision"
+        ] = tf.keras.mixed_precision.global_policy().name
+    else:
+        hyperparameters = None
+    return hyperparameters
+
+
+def _parse_model_history(model):
+    lines = None
+    if model.history is not None:
+        if model.history.history is not None:
+            lines = []
+            logs = model.history.history
+            num_epochs = len(logs["loss"])
+
+            for value in range(num_epochs):
+                epoch_dict = {
+                    log_key: log_value_list[value]
+                    for log_key, log_value_list in logs.items()
+                }
+                values = dict()
+                for k, v in epoch_dict.items():
+                    if k.startswith("val_"):
+                        k = "validation_" + k[4:]
+                    elif k != "epoch":
+                        k = "train_" + k
+                    splits = k.split("_")
+                    name = " ".join([part.capitalize() for part in splits])
+                    values[name] = v
+                lines.append(values)
+    return lines
+
+
+def _plot_network(model, save_directory):
+    tf.keras.utils.plot_model(
+        model,
+        to_file=f"{save_directory}/model.png",
+        show_shapes=False,
+        show_dtype=False,
+        show_layer_names=True,
+        rankdir="TB",
+        expand_nested=False,
+        dpi=96,
+        layer_range=None,
+        show_layer_activations=True,
+    )
+
+
+def _write_metrics(model, model_card):
+    lines = _parse_model_history(model)
+    if lines is not None:
+        model_card += "\n| Epochs |"
+
+        for i in lines[0].keys():
+            model_card += f" {i} |"
+        model_card += "\n |"
+        for i in range(len(lines[0].keys()) + 1):
+            model_card += "--- |"  # add header of table
+        for line in lines:
+            model_card += f"\n| {lines.index(line) + 1}|"  # add values
+            for key in line:
+                value = round(line[key], 3)
+                model_card += f" {value}| "
+    else:
+        model_card += "Model history needed"
+    return model_card
+
+
+def _create_model_card(
+    model,
+    repo_dir: Path,
+    model_plot: Optional[bool] = True,
```

Comment:
nit: maybe `plot_model` makes more sense here (verb instead of noun)

Reply:
I thought it would be very confusing because it's a variable that user asks for model's plot and there's a function called `_plot_model()` (though not seen in the user side) are you sure?

Reply:
fixed it with last commit!